### PR TITLE
Add stream names list

### PR DESCRIPTION
**Pull request by Jarema**

### CODE REVIEW

File: async-nats/src/jetstream/context.rs [hunk 1]
```diff
@@ -870,6 +903,82 @@ impl IntoFuture for PublishAckFuture {
...
 
+#[derive(Deserialize, Debug)]
+struct StreamPage {
+    total: usize,
+    streams: Option<Vec<String>>,
+}
+
+type PageRequest = Pin<Box<dyn Future<Output = Result<StreamPage, Error>>>>;
+
+pub struct StreamNamesList {
```
Comment (caspervonb) [inline async-nats/src/jetstream/context.rs:914]:
```suggestion
pub struct StreamNames {
```

File: async-nats/src/jetstream/context.rs [hunk 2]
```diff
@@ -465,6 +467,33 @@ impl Context {
...
+    /// let client = async_nats::connect("demo.nats.io:4222").await?;
+    /// let jetstream = async_nats::jetstream::new(client);
+    /// let mut names = jetstream.list_stream_names();
+    /// while let Some(stream) = names.try_next().await? {
+    ///     println!("stream: {}", stream);
+    /// }
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn list_stream_names(&self) -> StreamNamesList {
```
Comment (caspervonb) [inline async-nats/src/jetstream/context.rs:487]:
```suggestion
    pub fn stream_names(&self) -> StreamNamesList {
```

**Review by caspervonb — CHANGES_REQUESTED** [review]
Bikeshedding

**Review by piotrpio — APPROVED** [review]
LGTM

**Review by caspervonb — APPROVED** [review]
LGTM.

Remember to squash.

**Pull request merged**